app/org/sagebionetworks/bridge/play/controllers/AuthenticationController.java

```diff
@@ -102,7 +87,40 @@ public Result resetPassword() throws Exception {
         authenticationService.resetPassword(passwordReset);
         return okResult("Password has been changed.");
     }
-    
+
+    private Result signInWithRetry(final int retryCounter) throws Exception {
+
+        UserSession session = getSessionIfItExists();
+        if (session != null) {
+            setSessionToken(session.getSessionToken());
+            return okResult(new UserSessionInfo(session));
+        }
+
+        final JsonNode json = requestToJSON(request());
+        final SignIn signIn = parseJson(request(), SignIn.class);
+        final Study study = getStudyOrThrowException(json);
+        try {
+            Logger.info("User " + signIn.getUsername() + " signing in for study " + study.getIdentifier() + ".");
+            session = authenticationService.signIn(study, signIn);
+        } catch(ConsentRequiredException e) {
+            setSessionToken(e.getUserSession().getSessionToken());
+            throw e;
+        } catch(ConcurrentModificationException e) {
+            if (retryCounter > 0) {
+                Logger.info("User " + signIn.getUsername() +
+                        " is having a race condition with signing in for study " + study.getIdentifier() + "." +
+                        " Will retry after 1 second.");
+                // controller.signIn() 95% is < 1000 ms 
+                Thread.sleep(1000);
```

Comment:
1000ms sleep time seems a little high to me. This means if the initial sign-in fails due to the lock issue, users could be waiting a long time for their sign-in to complete.

The median and avg for signIn is < 500ms, which means on average, the lock would be released after 250ms. So I'd prefer either 250ms sleep time with more retries or 1000ms timeout with only 1 retry.